tests/unit/api/objectGet.js

```diff
@@ -384,4 +385,101 @@ describe('objectGet API', () => {
                 });
         });
     });
+
+    it('should return InvalidObjectState if trying to GET the object in cold storage', done => {
+        const testGetRequest = {
+            bucketName,
+            namespace,
+            objectKey: objectName,
+            headers: {},
+            url: `/${bucketName}/${objectName}`,
+        };
+        mdColdHelper.putBucketMock(bucketName, null, () => {
+            mdColdHelper.putObjectMock(bucketName, objectName, mdColdHelper.getArchiveArchivedMD(), () => {
+                objectGet(authInfo, testGetRequest, false, log, err => {
+                    assert.strictEqual(err.is.InvalidObjectState, true);
+                    done();
+                });
+            });
+        });
+    });
+
+    it('should return RestoreAlreadyInProgress if trying to GET the object in cold storage', done => {
```

Comment:
in aws, if the object is currently being restore, we still got `InvalidObjectState` error instead of `RestoreAlreadyInProgress` for get-object api. See below, for the same object that is currently being restored:
```
➜  aws s3api restore-object --bucket xinli-bucket-test1 --key file2 --restore-request Days=12 --profile aws

An error occurred (RestoreAlreadyInProgress) when calling the RestoreObject operation: Object restore is already in progress

➜  ~ aws s3api get-object --bucket xinli-bucket-test1 --key file2 --profile aws outfile

An error occurred (InvalidObjectState) when calling the GetObject operation: The operation is not valid for the object's storage class
```

Reply:
I think it is fine to comply with AWS on this, even if it is different from what we said in the design. I am making the according changes. Thanks!
See: https://github.com/scality/cloudserver/pull/4506/commits/3de2f0ca90d980a0221b9722c13e3ecfe1e33526

@RachedBenMustapha , does it make sense to you?

Reply:
Thank you!